lib/Sema/DerivedConformanceEquatableHashable.cpp

```diff
@@ -744,66 +745,159 @@ static Expr* integerLiteralExpr(ASTContext &C, int64_t value) {
   return integerExpr;
 }
 
-/// Returns a new assignment expression that combines the hash value of an
-/// expression into a variable.
-/// \p C The AST context.
-/// \p resultVar The variable into which the hash value will be combined.
-/// \p exprToHash The expression whose hash value should be combined.
-/// \return The expression that combines the hash value into the variable.
-static Expr* combineHashValuesAssignmentExpr(ASTContext &C,
-                                             VarDecl* resultVar,
-                                             Expr *exprToHash) {
-  // <exprToHash>.hashValue
-  auto hashValueExpr = new (C) UnresolvedDotExpr(exprToHash, SourceLoc(),
-                                                 C.Id_hashValue, DeclNameLoc(),
-                                                 /*implicit*/ true);
+/// Returns a new \c CallExpr representing
+///
+///   hasher.appending(hashable)
+///
+/// \param C The AST context to create the expression in.
+///
+/// \param hasher The base expression to make the call on.
+///
+/// \param hashable The parameter to the call.
+static CallExpr* createHasherAppendingCall(ASTContext &C,
+                                           Expr* hasher,
+                                           Expr* hashable) {
+  // hasher.appending(_:)
+  SmallVector<Identifier, 1> argNames{Identifier()};
+  DeclName name(C, C.Id_appending, argNames);
```

Comment:
Nitpick: You don't need a SmallVector for this; just pass `{Identifier()}` directly to DeclName's constructor. Or if you want to give it a name, declare it as a C array: `Identifier argNames[] = …`. (As you do below, I see.)